js/src/main/scala/org/widok/bindings/HTML.scala

```diff
@@ -79,8 +79,11 @@ object HTML {
 
   case class Button(contents: View*) extends Widget[Button] {
     val rendered = DOM.createElement("button", contents)
+    def buttonType(value: String) = attribute("type", value)
   }
 
+  def SubmitButton(contents: View*) = Button(contents: _*).buttonType("submit")
```

Comment:
I agree. You can create a ``sealed trait`` hierarchy with all possible button types.